src/history/test/HistoryTestsUtils.h

```diff
@@ -205,6 +205,22 @@ class CatchupSimulation
     std::vector<SequenceNumber> bobSeqs;
     std::vector<SequenceNumber> carolSeqs;
 
+    void ensurePublishesComplete();
```

Comment:
For verifying state I would just use 'check' or 'verify' names. For me 'ensure' means: check if something is true and if not, do everything to make sure it is true.

For example, `generateAndPublishHistory` was used to generate `n` history checkpoints. New methods, like `ensureOfflineCatchupPossible` are used to make sure that it will be possible to catchup to given ledger. It means that those may or may not generate new ledgers and/or publish them (it depends on current state of the simulation). And it is less iterative and more declarative it its naming, so it should be easier to understand. i.e.:

```
    catchupSimulation.ensureOnlineCatchupPossible(checkpointLedger);
    REQUIRE(catchupSimulation.catchupOnline(app, checkpointLedger));
```

Is much cleaner than previous:

```
    catchupSimulation.generateAndPublishInitialHistory(3);
    catchupSimulation.catchupNewApplication(
        catchupSimulation.getApp().getLedgerManager().getLastClosedLedgerNum() +
            1,
        std::numeric_limits<uint32_t>::max(), false,
        Config::TESTDB_IN_MEMORY_SQLITE, "s3");
```

Here you need to understand relationship between `getLastClosedLedgerNum` and catchup mechanism (and it is non-essential for this test), and also there is not-so-obvious code that adds extra ledgers inside `generateAndPublishInitialHistory` that is not needed for publish, but is needed for catchup. Naming the method `...CatchupPossible` makes it clear, that without those extra ledgers catchup would not be possible.

Also I'm using two functions - `ensureLedgerAvailable` and `ensurePublishesComplete` because it makes them doing only one thing and it allows for better naming. Those are always used together, but that does not mean that using one entity to do two things would be good.

Reply:
I don't feel super strongly about these (personally, just found it a bit hard to follow, and for `ensureLedgerAvailable` and `ensurePublishesComplete` functions I don't really understand what is the test case when we need to use one but not the other? ), so I'm not opposed to keeping it this way